packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/plugin-ledger-connector-fabric.ts

```diff
@@ -2013,4 +2016,48 @@ export class PluginLedgerConnectorFabric
 
     return discoveryResults;
   }
+
+  /**
+   * Creates a new Fabric event listener.
+   *
+   * @param req The request object containing the channel name and contract name.
+   * @param callback The callback function to be called when an event is received.
+   * @returns A promise that resolves to an object containing the listener and a
+   * function to remove it.
+   */
+  public async createFabricListener(
+    req: CreateListenerRequest,
+    callback: (event: ContractEvent) => Promise<void>,
+  ): Promise<{
+    removeListener: () => void;
+    listener: Promise<ContractListener>;
+  }> {
+    const fnTag = `${this.className}#createFabricListener()`;
+
+    const listener = async (contractEvent: ContractEvent) =>
+      callback(contractEvent);
+
+    try {
+      const gateway = await this.createGatewayWithOptions(req.gatewayOptions);
```

Comment:
@joaoecsde I've just checked that it is better to have the creation of the gateway using the `createGateway(...)` method instead of using `createGatewayWithOptions`. This allows creating the gateway with the`signingCredential` if `gatewayOptions` are not provided. Could you please update?

Reply:
As a suggestion, I assume we'll have to change the params of the `createGateway` function to something like `req: { gatewayOptions?: GatewayOptions, signingCredential?: FabricSigningCredential }`